### PR TITLE
Feature/default provider model

### PR DESCRIPTION
**Pull request by geclos**
Adds default provider with a limit of runs per customer

### CODE REVIEW

File: packages/core/src/services/commits/runDocumentAtCommit.test.ts [hunk 1]
```diff
@@ -29,54 +29,23 @@ describe('runDocumentAtCommit', () => {
...
-    })
-
-    await expect(result?.value?.response).rejects.toThrowError(
-      'Error validating document configuration',
-    )
-    await expect(result?.value?.duration).rejects.toThrowError(
-      'Error validating document configuration',
-    )
-  })
-
```
Comment (csansoon) [inline packages/core/src/services/commits/runDocumentAtCommit.test.ts:62]: It doesn't fail now?

File: apps/infra/src/core/elasticCache.ts [hunk 2]
```diff
@@ -25,14 +25,14 @@ const SecurityGroup = new aws.ec2.SecurityGroup('LatitudeLLMCacheSg', {
...
-//   securityGroupIds: [SecurityGroup.id],
-// })
+const cacheCluster = new aws.elasticache.Cluster('LatitudeLLMCacheCluster', {
+  engine: 'redis',
+  nodeType: 'cache.t3.micro',
+  numCacheNodes: 1,
+  port: 6379,
+  subnetGroupName: subnetGroup.name,
+  securityGroupIds: [SecurityGroup.id],
+})
```
Comment (csansoon) [inline apps/infra/src/core/elasticCache.ts:35]: Why are we enabling elastic now?

Comment (geclos) [reply apps/infra/src/core/elasticCache.ts:35]: this is not elasticsearch this is redis 😅 , we use it for our cache

Comment (geclos) [reply apps/infra/src/core/elasticCache.ts:35]: https://en.wikipedia.org/wiki/Amazon_ElastiCache

File: packages/core/src/services/ai/helpers.ts [hunk 3]
```diff
@@ -66,8 +66,8 @@ export function createProvider({
 export function validateConfig(config: Record<string, unknown>): Config {
   const configSchema = z
     .object({
-      model: z.string(),
-      provider: z.string(),
+      model: z.string().optional(), // TODO: make it not optional if we ever remove the default provider
+      provider: z.string().optional(), // TODO: make it not optional if we ever remove the default provider
```
Comment (andresgutgon) [inline packages/core/src/services/ai/helpers.ts:70]: This is not a todo, maybe a conditional todo xD

Comment (geclos) [reply packages/core/src/services/ai/helpers.ts:70]: good point

File: packages/sdks/typescript/rollup.config.mjs [hunk 4]
```diff
@@ -13,10 +13,6 @@ const aliasEntries = {
       find: '@latitude-data/core/browser',
       replacement: path.resolve(__dirname, '../../core/src/browser'),
     },
-    {
-      find: '@latitude-data/core/lib/Result',
-      replacement: path.resolve(__dirname, '../../core/src/lib/Result.ts'),
-    },
```
Comment (andresgutgon) [inline packages/sdks/typescript/rollup.config.mjs:19]: Why this?

File: apps/infra/src/core/elasticCache.ts (hunk 2 above)
Comment (andresgutgon) [reply apps/infra/src/core/elasticCache.ts:35]: Ah lol

File: packages/sdks/typescript/rollup.config.mjs (hunk 4 above)
Comment (geclos) [reply packages/sdks/typescript/rollup.config.mjs:19]: wasn't needed

**Pull request merged**